lib/twitter-ads/resources/dsl.rb

```diff
@@ -20,7 +20,14 @@ module InstanceMethods
       # @return [self] A fully hydrated instance of the current class.
       #
       # @since 0.1.0
-      def from_response(object)
+      def from_response(object, headers = nil)
+        if !headers.nil?
+          TwitterAds::Utils.extract_response_headers(headers).each { |key, value|
+            singleton_class.class_eval { attr_accessor key }
```

Comment:
My only gut reaction seeing singleton was that we should test it to make sure if there are multiple objects they aren't all sharing the same rate limit headers being over written from each other, but i think it's ok since each object would be a 'singleton' itself??

Reply:
@jbabichjapan Added a test case here: https://github.com/twitterdev/twitter-ruby-ads-sdk/pull/207/commits/c93cde71adea0e9841555b96ff628b008b9f7a96